kubernetes/windows/main.ps1

```diff
@@ -289,35 +289,99 @@ function Start-Fluent {
         (Get-Content -Path C:/etc/fluent/fluent.conf -Raw)  -replace 'fluent-docker-parser.conf','fluent-cri-parser.conf' | Set-Content C:/etc/fluent/fluent.conf
     }
 
+    # Start telegraf only in sidecar scraping mode
+    $sidecarScrapingEnabled = [System.Environment]::GetEnvironmentVariable('SIDECAR_SCRAPING_ENABLED')
+    if (![string]::IsNullOrEmpty($sidecarScrapingEnabled) -and $sidecarScrapingEnabled.ToLower() -eq 'true')
+    {
+        Write-Host "Starting telegraf..."
+        Start-Telegraf
+    }
+
     fluentd --reg-winsvc i --reg-winsvc-auto-start --winsvc-name fluentdwinaks --reg-winsvc-fluentdopt '-c C:/etc/fluent/fluent.conf -o C:/etc/fluent/fluent.log'
 
     Notepad.exe | Out-Null
 }
 
-function Generate-Certificates {
-    Write-Host "Generating Certificates"
-    C:\\opt\\omsagentwindows\\certgenerator\\certificategenerator.exe
-}
+function Start-Telegraf {
+    # Set default telegraf environment variables for prometheus scraping
+    Write-Host "**********Setting default environment variables for telegraf prometheus plugin..."
+    .\setdefaulttelegrafenvvariables.ps1
+
+    # run prometheus custom config parser
+    Write-Host "**********Running config parser for custom prometheus scraping**********"
+    ruby /opt/omsagentwindows/scripts/ruby/tomlparser-prom-customconfig.rb
+    Write-Host "**********End running config parser for custom prometheus scraping**********"
+
+
+    # Set required environment variable for telegraf prometheus plugin to run properly
+    Write-Host "Setting required environment variables for telegraf prometheus input plugin to run properly..."
+    $kubernetesServiceHost = [System.Environment]::GetEnvironmentVariable("KUBERNETES_SERVICE_HOST", "process")
+    if (![string]::IsNullOrEmpty($kubernetesServiceHost)) {
+        [System.Environment]::SetEnvironmentVariable("KUBERNETES_SERVICE_HOST", $kubernetesServiceHost, "machine")
```

Comment:
why  setting of machine wide envvar is required? afik. machine wide env can only set from DockerFile.

Reply:
> why setting of machine wide envvar is required? afik. machine wide env can only set from DockerFile.
> 
> Rashmi - Telegraf prometheus plugin needs env variable in the machine scope. Else it doesnt find it and wont work as expected.

> are you saying setting "machine" in main.ps1 wide env var is working?